cytoscape-matterjs.js

```diff
@@ -0,0 +1,428 @@
+            } else {
+              matterNodes.push({
+                shape:Bodies.circle(
+                  nodesIn[i].position().x,
+                  nodesIn[i].position().y,
+                  nodesIn[i].width() / 2,
+                  {
+                    id: nodesIn[i].data().id,
+                    inertia: Infinity,
+                    mass: 10,
+                    gravity: options.gravity,
+                    frictionAir: options.globalAirFriction,
+                    render: {
+                      strokeStyle: Common.shadeColor('#4ECDC4', -20),
+                      fillStyle: '#4ECDC4',
+                    }
+                  }
+                ),
+                parent: nodesIn[i].data().parent,
+                children: [],
+                oldX: nodesIn[i].position().x,
+                oldY: nodesIn[i].position().y
+              });
+              World.addBody(engine.world, matterNodes[currentMatterNode].shape);
+              addToCenterSprings(currentMatterNode);
+            }
+            for (var j = 0; j < nodesIn[i].children().length; j++) {
+              matterNodes[currentMatterNode].children[j] = nodesIn[i].children()[j].data().id;
+            }
+            currentMatterNode++;
+          }
+        }
+        var xAvg = 0;
+        var yAvg = 0;
+        for (var i = 0; i < options.clusters.length; i++) {
+          xAvg = 0;
+          yAvg = 0;
+          for (var j = 0; j < options.clusters[i].nodes.length; j++) {
+            for (var k = 0; k < nodes.length; k++) {
+              if (nodes[k].data().id === options.clusters[i].nodes[j]) {
+                xAvg += nodes[k].position().x;
+                yAvg += nodes[k].position().y;
+              }
+            }
+          }
+          for (var j = 0; j < options.clusters[i].nodes.length; j++) {
+            for (var k = 0; k < matterNodes.length; k++) {
+              if (matterNodes[k].shape.id === options.clusters[i].nodes[j]) {
+                World.add(world, Constraint.create({
+                  bodyA: matterNodes[k].shape,
+                  pointB: { x: xAvg/options.clusters[i].nodes.length, y: yAvg/options.clusters[i].nodes.length },
+                  length: options.clusters[i].distanceFromCluster,
+                  stiffness: options.clusters[i].elasticityToCluster
+                }));
+              }
+            }
+          }
+        }
+        return 0;
+      }
+      mapNodes(nodes);
+
+// +------------------------------------------------------------------------+ //
+// +----------------ADDS EDGES TO THE MATTER JS SIMULATION------------------+ //
+// +------------------------------------------------------------------------+ //
+// +------------------------------------------------------------------------+ //
+
+      function mapEdges(edgesIn){
+        for (var i = 0; i < edgesIn.length; i++) {
+          var tempSource;
+          var tempTarget;
+          for (var j = 0; j < matterNodes.length; j++) {
```

Comment:
Each `node` should store a ref to its `matterNode`, so you can do a fast lookup.